src/App.tsx

```diff
@@ -1,26 +1,144 @@
-/* eslint-disable max-len */
+/* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+import { Error, Footer, NewTodo, TodoItem, TodoList } from './components';
 import { UserWarning } from './UserWarning';
-
-const USER_ID = 0;
+import { USER_ID, getTodos, deleteTodo, createTodo } from './api/todos';
+import { Todo } from './types/Todo';
+import { FilterOptions } from './types/FilterOptions';
 
 export const App: React.FC = () => {
+  const [currentTodoList, setCurrentTodoList] = useState<Todo[]>([]);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [activeFilter, setActiveFilter] = useState(FilterOptions.ALL);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [isInputDisabled, setIsInputDisabled] = useState(false);
+  const [inputValue, setInputValue] = useState('');
+  const [todosBeingDeleted, setTodosBeingDeleted] = useState<number[]>([]);
+
+  const handleCreateTodo = async ({
+    title,
+    userId,
+    completed,
+  }: Omit<Todo, 'id'>) => {
+    const tempTodoItem: Todo = { id: 0, title, userId, completed };
+
+    setTempTodo(tempTodoItem);
+
+    setIsInputDisabled(true);
+
+    try {
+      const newTodo = await createTodo({ title, userId, completed });
+
+      setCurrentTodoList(currentTodos => [...currentTodos, newTodo]);
+      setTempTodo(null);
+      setInputValue('');
+    } catch (error) {
+      setErrorMessage('Unable to add a todo');
+      setTimeout(async () => setErrorMessage(''), 3000);
```

Comment:
can we have useEffect in Error component that handle timer and error 
that way you shouldn't handle it in every function DRY